tests/trainer/test_dataloaders.py

```diff
@@ -899,6 +899,24 @@ def gen(self):
     assert trainer.current_epoch == 1
 
 
+def test_iterable_dataset_stop_iteration_at_start():
+    """Test that the training loop skips execution if the iterator is empty."""
+
+    class RandomDataset(IterableDataset):
+        def __iter__(self):
+            return iter(self.gen())
+
+        def gen(self):
+            yield from []
+
+    model = BoringModel()
+    train_dataloader = DataLoader(RandomDataset(), batch_size=2)
+    trainer = Trainer(default_root_dir=os.getcwd(), max_epochs=2)
+    trainer.fit(model, train_dataloader=train_dataloader)
+    assert trainer.global_step == 0
+    assert trainer.current_epoch == 1
```

Comment:
Not sure if this value is correct, two arguments:

- No batch was run so current epch should be `0`
- An epoch composed of 0 batches was successfully run, the current epoch should be `1`

`1` is the value in master

Reply:
hey, when I fixed this specific edge case I added a test for it: 

`test_iterable_dataset_stop_iteration_at_epoch_beginning`

it may be simpler to adapt this one. whether the iterator is empty from the beginning of the training or the beginning of an epoch should not matter

Reply:
Merged